src/neo/SmartContract/InteropService.Contract.cs

```diff
@@ -40,44 +41,46 @@ private static bool Contract_Create(ApplicationEngine engine)
                 contract = new ContractState
                 {
                     Id = engine.Snapshot.ContractId.GetAndChange().NextId++,
-                    Script = script,
+                    Script = script.ToArray(),
                     Manifest = ContractManifest.Parse(manifest)
                 };
 
                 if (!contract.Manifest.IsValid(hash)) return false;
 
                 engine.Snapshot.Contracts.Add(hash, contract);
-                engine.CurrentContext.EvaluationStack.Push(StackItem.FromInterface(contract));
+                engine.Push(StackItem.FromInterface(contract));
                 return true;
             }
 
             private static bool Contract_Update(ApplicationEngine engine)
             {
-                byte[] script = engine.CurrentContext.EvaluationStack.Pop().GetSpan().ToArray();
-                if (script.Length > 1024 * 1024) return false;
-                var manifest = engine.CurrentContext.EvaluationStack.Pop().GetString();
-                if (manifest.Length > ContractManifest.MaxLength) return false;
+                if (!engine.TryPop(out StackItem item0)) return false;
```

Comment:
`scriptItem` and `manifestItem` ?

Reply:
I think it's okay because `item0` means the top item on the stack.